### DISCUSSION TITLE
Why are issues locked even after community feedback clearly shows issues as not being resolved?

### DISCUSSION THREAD

**Discussion by garth** [vercel/next.js#54878] [Help] [answered]
### Summary

@timneutkens closed https://github.com/vercel/next.js/issues/49578 when, if you read the comments, it's still a real blocking issue that has not been fixed. But there is no way to continue to conversation and and work together to find a solution when this happens, especially when some bot the locks the discussion.

Unfortunately this kind of action just alienates people.

**Comment by timneutkens** [comment]
I understand that it can be frustrating when an issue is closed, you're right that it would be awesome if people could keep posting comments on issues and that would lead to some kind of resolution. However, those comments should then lead to something actionable from either the people involved on the issue or from our team.

In order to ensure that the discussion stays actionable we ask people to open a new issue so that we can triage and support whichever issue being run into.
Often times when a new issue is opened out of an existing issue it turns out the report is for an entirely different issue with a similar error / error code and needs an entirely separate bugfix, sometimes it's even just upgrading Next.js as the application is on an old version of Next.js.

In case of the issue you linked for example: fetch failed can happen in your own code when the url fetched is non-existent, or if you try to fetch API Routes / Route Handlers during static generation. Unfortunately the fetch spec doesn't give you much to go on with `fetch` failing. So in order for anyone to be helpful they would need to see the application in order to pinpoint what is causing it.

The additional comments of "still happens" are not enough to investigate further as you can imagine. 
This is why we ask for a reproduction in a new issue as it's often unclear to people what to provide to be helpful and the issue template guides you through providing all required details for us to investigate and provide support. 

In order to effectively maintain this repository which gets over 200 new issues opened per month and hundreds of comments per week we have to ensure that the information shared is actionable, which is why we recommend opening a new issue as it allows the team to effectively verify and triage issues, currently that is done by 1 person full time (@balazsorban44) and we've hired a second person to help with triaging who is starting soon. Since the triaging process is very time consuming by itself, especially when there is no reproduction provided, we have to guide people towards creating new issues with a reproduction.

Since you mentioned that you're still running into something similar to #49578 we'd be more than happy to investigate if you provide a reproduction so that we can verify it's the same or something else.

Our intention with this process is making sure that we're able to support you (and hundreds of thousands of others) adequately with a small team (8 people full time currently).

**Reply by garth** [reply]
I agree with almost all the points you highlighted, even marking as issues as being closed. Of course the community understands the scale of the task of maintaining a project of the scale of nextjs. But as far as I can see, marking the issue as locked has no benefit to your team?

Leaving the issue unlocked would allow the community to continue to work together, which might allow for additional insight which cannot happen with locked issues.

Also consider than the barrier for submitting new issues is very high. I am always extremely grateful when another member of the community takes the time and effort to document and provide instructions for a reproduction. When forced to open a new issue, even if the majority of participants are still affected, they may not be in a position to do so, and even if they do, the discussion is now fragmented, often making it harder to piece together all the clues that may be necessary to find a resolution.

**Reply by timneutkens** [reply, marked as answer]
The happy path that you outlined would be awesome. Unfortunately our experience has been quite different when not locking issues, specifically what happened on most issues where people would end up posting "Same issue" after it's closed is they would end up in one of these situations:
- People get incredibly frustrated (including swearing etc) and hostile towards maintainers for not re-opening the issue. For example I've received some frankly unacceptable messages with threats the past few months from anonymous developers on Twitter.
- If we did post a reply saying the comments are not helpful, asking for a reproduction, most replies would be like "There already is a reproduction you figure it out" but that reproduction was already fixed as that's why the issue is closed, and explaining that it's already fixed is then often met with something equivalent to "But I still run into it, fix it" instead of providing a reproduction.
- People coming back 1 / 2 years later saying "Same issue" on something they can't possibly be running into it

These additional comments take a significant amount of effort and overhead to get into a place where they are useful, this is even a struggle on issues that are open, which is time that could be spent on fixing bugs / supporting requests that do have reproductions.

Basically not everyone is as willing to help get to a resolution as you are. In my opinion that is totally fine.

It just requires a bit more elaborate process to ensure people create issue with a reproduction instead of drive-by comments, which includes not being able to post the drive-by comment in the first place so that you have to take the required steps as the issue tracker is where we actively work.

In terms of general discussion our recommendation is leveraging GitHub Discussion, e.g. when the issue is closed and you don't want to spend time creating a reproduction you could open a GitHub discussion to try and collaborate with the community, without needing intervention from our team until there is a reproduction. GitHub discussions are always kept open (unless you mark something as an answer).

**Reply by garth** [reply]
Thank you for taking the time make such a detailed response... I'm sorry that you are subjected to such unpleasant comments at times. The vast majority very much appreciate the work that you do.

**Reply by timneutkens** [reply]
No problem at all! If you're still running into something like https://github.com/vercel/next.js/issues/49578 we'd be happy to investigate that further on a new issue 🙌
